### PR TITLE
Binary outcome variable in IRM and IIVM

### PR DESCRIPTION
**Pull request by PhilippBach**
- Use classifiers for binary outcome in IRM and IIVM models
- Tests for binary IRM and IIVM model
- ~~Implement a DGP for binary outcome IRM (not 100% sure if we really want to include the DGP)~~

### CODE REVIEW

File: R/double_ml_iivm.R [hunk 1]
```diff
@@ -129,9 +138,11 @@ DoubleMLIIVM = R6Class("DoubleMLIIVM",
...
     #' of the causal model.
     #'
-    #' @param ml_g ([`LearnerRegr`][mlr3::LearnerRegr], `character(1)`) \cr
-    #' An object of the class [mlr3 regression learner][mlr3::LearnerRegr] to
-    #' pass a learner, possibly with specified parameters, for example
+    #'  @param ml_g ([`LearnerRegr`][mlr3::LearnerRegr],
+    #' [`LearnerClassif`][mlr3::LearnerClassif], `character(1)`,) \cr
+    #' An object of the class [mlr3 regression learner][mlr3::LearnerRegr] or
+    #' [mlr3 classification learner][mlr3::LearnerClassif] to pass a learner,
+    #' possibly with specified parameters, for example
```
Comment (MalteKurz) [inline R/double_ml_iivm.R:145]: Can we really just use a classifier for `g(D, X)` without any additional requirements on `Y`? I would have expected an additional requirement like binary and values 0 and 1.

Comment (PhilippBach) [reply R/double_ml_iivm.R:145]: Thanks, I have a look at it again.

Comment (PhilippBach) [reply R/double_ml_iivm.R:145]: So currently, we are checking the levels of the outcome (in any nuisance part) during instantiation of the underlying prediction task in `mlr3`.

https://github.com/DoubleML/doubleml-for-r/blob/a10e38d5e70eacf85742976c5e92ca03abcf41c3/R/helper.R#L232-L234

Using other levels for the outcome (here `y`), will raise the error (I replaced the binary 1/0 outcome with a 2/0 outcome): 
```
Error in initiate_task(id = nuisance_id, data = data_model, target = y_col,  : 
  Assertion on 'levels(data[[target]])' failed: Must be equal to set {'0','1'}, but is {'0','2'}.
```

So to answer you question. We do check for the levels of the outcome - but at a fairly late stage (during calling `fit()`), we could in general do this also during instantiation of the learners /  `DoubleMLIRM` and `DoubleMLIIVM` model object, respectively (the same is true for the binary treatment variable in a `DoubleMLPLR`)

File: tests/testthat/helper-04-simdata.R [hunk 2]
```diff
@@ -5,8 +5,10 @@ setting = list(theta = 0.5, n = 1000, p = 20)
 on_cran = !identical(Sys.getenv("NOT_CRAN"), "true")
 if (on_cran) {
   setting_irm = list(theta = 0.5, n = 1000, p = 20)
+  setting_irm_binary = list(theta = 0.5, n = 1000, p = 20)
 } else {
   setting_irm = list(theta = 0.5, n = 5000, p = 20)
+  setting_irm_binary = list(theta = 0.5, n = 5000, p = 20)
```
Comment (MalteKurz) [inline tests/testthat/helper-04-simdata.R:11]: Do we really need this alternative `setting_irm_binary` variable if it is anyways in both cases equal to `setting_irm`?

**Review by MalteKurz — APPROVED** [review]
LGTM, I only have one minor comment with regards to a unit test.

File: tests/testthat/helper-04-simdata.R (hunk 2 above)
Comment (PhilippBach) [reply tests/testthat/helper-04-simdata.R:11]: I agree, we don't need the additional setting. Thanks!

**Pull request merged**